$OpenBSD: patch-qutebrowser_misc_elf_py,v 1.2 2021/04/07 16:12:49 sthen Exp $

Fix QtWebEngine version detection on OpenBSD, similar to / replaced by

From a38fabda2581790d365cb82ea37cf45c1c7bb3a6 Mon Sep 17 00:00:00 2001
From: Dimitri Karamazov <deserter666@danwin1210.me>
Date: Tue, 6 Apr 2021 23:13:43 +0530
Subject: [PATCH] Remove reduntant operations

Index: qutebrowser/misc/elf.py
--- qutebrowser/misc/elf.py.orig
+++ qutebrowser/misc/elf.py
@@ -316,8 +316,8 @@ def parse_webenginecore() -> Optional[Versions]:
     else:
         library_path = pathlib.Path(QLibraryInfo.location(QLibraryInfo.LibrariesPath))
 
-    # PyQt bundles those files with a .5 suffix
-    lib_file = library_path / 'libQt5WebEngineCore.so.5'
+    library_name = sorted(list(pathlib.Path(library_path).glob('libQt5WebEngineCore.so*')))
+    lib_file = library_path / library_name[-1].as_posix()
     if not lib_file.exists():
         return None
 
